src/pages/AddPatient.jsx

```diff
@@ -15,33 +15,29 @@ import { useState } from "react";
 import Toast from "../components/Toast";
 const AddPatient = ({ open, onClose }) => {
   const { register, handleSubmit, reset } = useForm();
-  const [openToast, setOpenToast] = useState(false);
-  const [toastMessage, setToastMessage] = useState("");
-  const [toastSeverity, setToastSeverity] = useState("");
+  const [toastData, setToastData] = useState({
+    openToast: false,
+    toastMessage: "",
+    toastSeverity: ""
+  });
+  const updateToastData = (key, value) => {
```

Comment:
No need for a separate function here, just calling setToastData with the changes is how it is normally done
And you are usually updating all 3 properties at the same time so just pass in a new object.